aea/aea.py

```diff
@@ -297,11 +297,11 @@ def _handle_decoding(
             return None, []  # Tuple[Optional[Message], List[Handler]]
 
         handlers = self.filter.get_active_handlers(
-            protocol.public_id, envelope.to_as_public_id or envelope.skill_id
+            protocol.public_id, envelope.to_as_public_id
         )
 
         if len(handlers) == 0:
-            handler.send_unsupported_skill(envelope, self.logger)
+            handler.send_unsupported_handler(envelope, self.logger)
```

Comment:
probably unsupported protocol id/target?
what about checking target explicitly to provide an error like target skill does not present?
in case it's a check by the protocol id, we can inform that the protocol id is not supported by the agent.


also, not sure, do we have an agent2agent check for the target address match with current agent address?

Reply:
the address check happens automatically inside the skills' dialogues.

Reply:
+1 on `send_unsupported_protocol`.

Reply:
the correct cause is `no_active_handler`, updated now